dali/python/nvidia/dali/pipeline.py

```diff
@@ -907,11 +908,13 @@ def feed_input(self, data_node, data, layout=None, cuda_stream=None, use_copy_ke
             name = data_node.name
 
         # Check for use of feed_input on an external_source operator that was
-        # initialized with 'source'.
-        if next((op._callback is not None for op in self._ops if op.name == name), False):
-            raise RuntimeError(
-                f"Cannot use `feed_input` on the external source '{name}' with a `source`"
-                " argument specified.")
+        # initialized with 'source'. This check makes sense only for fully Python-based
+        # pipelines, and not deserialized ones
+        if not self._deserialized:
```

Comment:
I don't like the idea of making deserialized pipelines somehow worse and cutting them off some capabilities "because we know they can't" instead of checking for actual presence of absence of a given capability.
I'd go with:
```suggestion
        if self._ops is not None:
```
or even combining this line with the one below into one property `has_source` - because ultimately that's what we want to know. The fact that a serialized pipeline _currently_ doesn't store source callbacks seems coincidental - and we could change that with relative ease by pickling the source.

Reply:
My intention here was to partially revert the #3574 for the deserialized pipelines - with such check we can basically mark the problematic places and adjust them when we have proper functionality in place - instead of going into implementation details for it. 
`has_source` would be nice, but again, that just pushes the check further into DALI.

Reply:
For me, both solutions make equal sense. 
 I understand the reasoning behind Michal's solution and `if self._ops is not None:` looks good. 
 Also, I like the argument about using `._deserialized` which kind of documents that the problem is connected to the deserialization and such place can be revisited whenever we change the way we deal with deserializing.
 I think the PR is ok the way it is now, so I don't request any changes.